src/gui/editors/TimeLineWidget.cpp

```diff
@@ -225,83 +212,138 @@ void TimeLineWidget::paintEvent( QPaintEvent * )
 		}
 	}
 
-	// Draw the main rectangle (outer border)
+	// Draw the loop rectangle's outer outline
 	p.setPen( loopPointsActive ? getActiveLoopColor() : getInactiveLoopColor() );
 	p.setBrush( Qt::NoBrush );
 	p.drawRect( outerRectangle );
 
-	// Draw the inner border outline (no fill)
+	// Draw the loop rectangle's inner outline
 	QRect innerRectangle = outerRectangle.adjusted( 1, 1, -1, -1 );
 	p.setPen( loopPointsActive ? getActiveLoopInnerColor() : getInactiveLoopInnerColor() );
 	p.setBrush( Qt::NoBrush );
 	p.drawRect( innerRectangle );
+	
+	// Draw loop handles if necessary
+	const auto handleMode = ConfigManager::inst()->value("app", "loopmarkermode") == "handles";
+	const auto hw = std::min(m_loopHandleWidth, loopRectWidth / 2 - 1);
+	const auto leftHandle = QRectF(loopStart - .5, loopRectMargin - .5, hw, loopRectHeight);
+	const auto rightHandle = QRectF(loopEndR - hw - .5, loopRectMargin - .5, hw, loopRectHeight);
+	if (handleMode && underMouse() && QGuiApplication::keyboardModifiers().testFlag(Qt::ShiftModifier))
+	{
+		auto color = loopPointsActive ? m_activeLoopHandleColor : m_inactiveLoopHandleColor;
```

Comment:
Minor since it's a short lived variable, but would it be good practice to `const` this, too?

Reply:
Absolutely - `const` all the things.